Update async waitUntil microtask tests

aliams opened the pull request:
Fixed the async waitUntil microtask tests to reflect the spec. Please refer to https://github.com/w3c/ServiceWorker/issues/1213#issuecomment-342640579.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

Commit:
Update async waitUntil microtask issue

Fixed the async waitUntil microtask tests to reflect the spec

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/22439)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/325642933)

Commit:
Update async waitUntil microtask tests

Commit:
Add test to sw extendableevent microtasks

This will test calling waitUntil in the same microtask checkpoint after all the promises have settled. It is distinct in that the dispatch flag would have been unset.

jungkees commented:
@aliams, I left a follow-up comment on https://github.com/w3c/ServiceWorker/issues/1213#issuecomment-358293981.

jungkees commented on service-workers/service-worker/resources/extendable-event-async-waituntil.js:
```diff
@@ -45,6 +45,16 @@ self.addEventListener('message', function(event) {
       case 'script-extendable-event':
         self.dispatchEvent(new ExtendableEvent('nontrustedevent'));
         break;
+      case 'no-current-extension-different-microtask-after-dispatch-flag-unset':
```
It seems more like 'current-extension-different-microtask-after-dispatch-flag-unset' as `event.waitUntil(waitPromise)` is called in the event handler itself?

aliams replied:
Good point, I just changed it.

Commit:
Update to remove 'no' from test case
